### DISCUSSION TITLE
Why is the deformation gradient stored in vector fields in bond associated correspondence material?

### DISCUSSION THREAD

**Discussion by andrisfreimanis** [peridigm/peridigm#259] [Q&A] [answered]
Dear All,
I'm looking through Peridigm_BondAssociatedCorrespondenceMaterial.cpp and see that the node level deformation gradient is stored in three vector fields rather than a single tensor field. I can't think of any limitation or advantage to do that. There's probably something I'm missing here. Does anyone know why is it stored like that?

Best
Andris Freimanis

**Comment by sahirbutt** [comment, marked as answer]
I think, it has to do with the synchronization across the mpi and block boundaries, which is possible at the moment only with scalar and vector fields.

**Reply by oldninja** [reply]
I recall something like that too, but I'm not 100% sure. There should be an old thread or issue about this, where I recall that explanation from.

**Reply by andrisfreimanis** [reply]
Yep, that makes sense. Thanks for help.
